libexecutive/Executive.cpp

```diff
@@ -731,6 +713,94 @@ void Executive::revert()
     memoryTableFactory->rollback(m_tableFactorySavepoint);
 }
 
+void Executive::parseEVMCResult(std::shared_ptr<eth::Result> _result)
+{
+    auto outputRef = _result->output();
+    switch (_result->status())
+    {
+    case EVMC_SUCCESS:
+    {
+        m_gas = _result->gasLeft();
+        m_output = owning_bytes_ref(
+            bytes(outputRef.data(), outputRef.data() + outputRef.size()), 0, outputRef.size());
+    }
+    case EVMC_REVERT:
+    {
+        // FIXME: Copy the output for now, but copyless version possible.
+        m_gas = _result->gasLeft();
+        revert();
+        m_output = owning_bytes_ref(
+            bytes(outputRef.data(), outputRef.data() + outputRef.size()), 0, outputRef.size());
+        m_excepted = TransactionException::RevertInstruction;
+    }
+    case EVMC_OUT_OF_GAS:
+    case EVMC_FAILURE:
+    {
+        revert();
+        m_excepted = TransactionException::OutOfGas;
+    }
+
+    case EVMC_INVALID_INSTRUCTION:  // NOTE: this could have its own exception
+    case EVMC_UNDEFINED_INSTRUCTION:
+    {
+        m_gas = 0;
+        m_excepted = TransactionException::BadInstruction;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'm_excepted' is reassigned a value before the old one has been used. 'break;' missing?](https://app.codacy.com/gh/FISCO-BCOS/FISCO-BCOS/pullRequest?prid=5654262)